tensorflow_addons/metrics/multilabel_metrics_test.py

```diff
@@ -0,0 +1,181 @@
+            activation=_activation,
+        )
+        self.assertEqual(obj_a._name, _name)
+
+        # Check save and restore config
+        obj_b = _class.from_config(obj_a.get_config())
+        self.assertEqual(obj_b._name, _name)
+
+    def initialize_vars(self):
+        _class = MultiLabelMacroSensitivity
+        obj = _class()
+        self.evaluate(tf.compat.v1.variables_initializer(obj.variables))
+        return obj
+
+    def update_obj_states(self, obj, y_true, y_pred):
+        update_op = obj.update_state(y_true, y_pred)
+        self.evaluate(update_op)
+
+    def check_results(self, obj, value):
+        self.assertAllClose(value, self.evaluate(obj.result()), atol=1e-5)
+
+    def test_sensitivity_score(self):
+        obj = self.initialize_vars()
+        y_true = tf.cast(
+            tf.constant(
+                [[0, 0, 1, 1, 1], [0, 0, 1, 0, 1], [1, 0, 0, 0, 1],], dtype=tf.float32
+            ),
+            dtype=tf.float32,
+        )
+
+        y_pred = tf.cast(
+            tf.constant(
+                [[0, 0, 1, 0, 0], [1, 0, 1, 1, 0], [1, 0, 0, 0, 1],], dtype=tf.float32
+            ),
+            dtype=tf.float32,
+        )
+        self.update_obj_states(obj, y_true, y_pred)
+        self.check_results(obj, 0.5714286)
+
+
+@test_utils.run_all_in_graph_and_eager_modes
+class MultiLabelMacroSpecificityTest(tf.test.TestCase):
+    def test_config(self):
+        _class = MultiLabelMacroSpecificity
+        _name = "multi_label_macro_specificity"
+        _threshold = 0.5
+        _from_logits = True
+        _activation = "sigmoid"
+
+        obj_a = _class(
+            name=_name,
+            threshold=_threshold,
+            from_logits=_from_logits,
+            activation=_activation,
+        )
+        self.assertEqual(obj_a._name, _name)
+
+        # Check save and restore config
+        obj_b = _class.from_config(obj_a.get_config())
+        self.assertEqual(obj_b._name, _name)
+
+    def initialize_vars(self):
+        _class = MultiLabelMacroSpecificity
+        obj = _class()
+        self.evaluate(tf.compat.v1.variables_initializer(obj.variables))
+        return obj
+
+    def update_obj_states(self, obj, y_true, y_pred):
+        update_op = obj.update_state(y_true, y_pred)
+        self.evaluate(update_op)
+
+    def check_results(self, obj, value):
+        self.assertAllClose(value, self.evaluate(obj.result()), atol=1e-5)
+
+    def test_specificity_score(self):
+        obj = self.initialize_vars()
+        y_true = tf.cast(
+            tf.constant(
+                [[0, 0, 1, 1, 1], [0, 0, 1, 0, 1], [1, 0, 0, 0, 1],], dtype=tf.float32
+            ),
+            dtype=tf.float32,
+        )
+
+        y_pred = tf.cast(
+            tf.constant(
+                [[0, 0, 1, 0, 0], [1, 0, 1, 1, 0], [1, 0, 0, 0, 1],], dtype=tf.float32
+            ),
+            dtype=tf.float32,
+        )
+        self.update_obj_states(obj, y_true, y_pred)
+        self.check_results(obj, 0.75)
+
```

Comment:
Could you add tests for each metric which shows that the metric works correctly in a keras models for with several different label/loss types. See Cohen's Kappa as an example:
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/cohens_kappa_test.py#L160-#L228

Reply:
It doesn't. I was designed for specifically the case of having `n` labels and a dimension for the `n` labels:

e.g.

```
# labels 1 and 3
[0, 1, 0, 1]
```
rather than any of the following:
```
[1,3]
"1,3"
7 # if all combinations of labels e.g. 1,3 are embedded as an integer 
```

etc